azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AuthHelper.java

```diff
@@ -89,46 +71,14 @@ public static boolean validateEnvironment(String environment) {
      * @param environment the environment key
      * @return the AzureEnvironment instance
      */
-    public static AzureEnvironment parseAzureEnvironment(String environment) {
-        if (StringUtils.isEmpty(environment)) {
-            return null;
-        }
-
-        switch (preprocessAzureEnvironment(environment)) {
-            case AZURE_CHINA:
-                return AzureEnvironment.AZURE_CHINA;
-            case AZURE_GERMANY:
-            case AZURE_GERMAN: // the TYPO comes from azure cli: https://docs.microsoft.com/en-us/azure/germany/germany-get-started-connect-with-cli
-                return AzureEnvironment.AZURE_GERMANY;
-            case AZURE_US_GOVERNMENT:
-            case AZURE_US_GOVERNMENT2:
-                return AzureEnvironment.AZURE_US_GOVERNMENT;
-            default:
-                return AzureEnvironment.AZURE;
-        }
+    public static AzureEnvironment stringToAzureEnvironment(String environment) {
+        String targetEnvironment = StringUtils.replaceChars(environment, '-', '_');
+        return AZURE_CLOUD_ALIAS_MAP.entrySet().stream().filter(entry -> StringListUtils.containsIgnoreCase(Arrays.asList(entry.getValue()), targetEnvironment))
+                .map(Map.Entry::getKey)
+                .findFirst().orElse(null);
     }
 
-    private static String preprocessAzureEnvironment(String environment) {
-        if (StringUtils.isEmpty(environment)) {
-            return environment;
-        }
-        String formattedEnvironment = StringUtils.trim(StringUtils.lowerCase(environment));
-        if (StringUtils.equals(environment, AZURE)) {
-            return AZURE;
-        }
-        for (String startToRemove : Arrays.asList("azure-", "azure_", "azure")) {
-            if (StringUtils.startsWith(formattedEnvironment, startToRemove)) {
-                formattedEnvironment = StringUtils.removeStart(formattedEnvironment, startToRemove);
-                break;
-            }
-        }
-        for (String endToRemove : Arrays.asList("-cloud", "_cloud", "cloud")) {
-            if (StringUtils.endsWith(formattedEnvironment, endToRemove)) {
-                formattedEnvironment = StringUtils.removeEnd(formattedEnvironment, endToRemove);
-                break;
-            }
-        }
-
-        return formattedEnvironment;
+    private static void putAliasMap(AzureEnvironment env, String... aliases) {
+        AZURE_CLOUD_ALIAS_MAP.put(env, aliases);
```

Comment:
this method seems can be removed

Reply:
I don't want too many new String[] {"""}